src/cert_utils.erl

```diff
@@ -0,0 +1,86 @@
+-module(cert_utils).
+-author('denys.gonchar@erlang-solutions.com').
+
+-export([ get_cert_domains/1,
+          get_common_name/1,
+          get_xmpp_addresses/1
+        ]).
+
+-include_lib("public_key/include/public_key.hrl").
+-include("XmppAddr.hrl").
+-include("jlib.hrl").
+
+-type certificate() :: #'Certificate'{}.
+
+-spec get_common_name(certificate()) -> [].
+get_common_name(Cert) ->
+  try
+    {rdnSequence, RDNSequence} = Cert#'Certificate'.tbsCertificate#'TBSCertificate'.subject,
+    [{ok, {_, CN}}] = ['OTP-PUB-KEY':decode('X520CommonName', V) ||
+      AtributesList <- RDNSequence,
+      #'AttributeTypeAndValue'{type = ?'id-at-commonName', value = V} <- AtributesList],
+    CN
+  catch
+    _:_ -> error
+  end.
+
+
+-spec get_xmpp_addresses(certificate()) -> [any()].
+get_xmpp_addresses(Cert) ->
+  try
+    Extensions = Cert#'Certificate'.tbsCertificate#'TBSCertificate'.extensions,
+    [BinVal] = [convert_to_bin(V) ||
+      #'Extension'{extnID = ?'id-ce-subjectAltName', extnValue = V} <- Extensions],
+    {ok, SANs} = 'OTP-PUB-KEY':decode('SubjectAltName', BinVal),
+    [begin
+       {ok, XmppAddr} = 'XmppAddr':decode('XmppAddr', V),
+       XmppAddr
+     end || {otherName, #'AnotherName'{'type-id' = ?'id-on-xmppAddr', value = V}} <- SANs]
+  catch
+    _:_ -> []
```

Comment:
report an error here

Reply:
i'm not gonna log here anything, it's normal when certificate is missing some of the fields, generally  it's not an error. if you want to log something - do it in the proper place, it's a duty of the caller

Reply:
Than this code can be rewritten without try ... catch.
Just by expecting missing fields.
There also should be at least two tests: one with missing field, one with not.

.... ideally.

Reply:
It deserves a separate error return, when a field expected to have a valid XMPP address, does not actually have one. If fields are missing at all, then I think it's OK to return `[]`.